### PR TITLE
fix: handle null values for braze dedupe

### PR DESCRIPTION
**Pull request by utsabc**
## What are the changes introduced in this PR?

This PR approaches to fix the `null` values handling during dedupe. When a property is set to as `null` it is unset from braze hence when we lookup it is not available. Hence we need to handle that scneario during deduplication

## What is the related Linear task?

Resolves INT-2498

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3638/test-report.html

File: src/v0/destinations/braze/braze.util.test.js [hunk 1]
```diff
@@ -785,6 +785,136 @@ describe('dedup utility tests', () => {
...
+          age: 30,
+        },
+      };
+      store.set('123', storeData);
+      const result = BrazeDedupUtility.deduplicate(userData, store);
+      expect(store.size).toBe(1);
+      expect(result).toEqual(null);
+    });
+
+    test('deduplicates user data correctly when user data is undefined and it is same in stored data', () => {
```
Comment (krishna2020) [inline src/v0/destinations/braze/braze.util.test.js:895]:
```suggestion
    test('deduplicates user data correctly when user data is undefined and it is defined in stored data', () => {
```

**Review by krishna2020 — DISMISSED** [review]

File: src/v0/destinations/braze/util.js [hunk 2]
```diff
@@ -288,7 +288,12 @@ const BrazeDedupUtility = {
 
     if (keys.length > 0) {
       keys.forEach((key) => {
-        if (!_.isEqual(userData[key], storedUserData[key])) {
+        // in case of keys having null values only compare if the key is present in the stored user data
+        if (userData[key] === null) {
```
Comment (koladilip) [inline src/v0/destinations/braze/util.js:292]:
```
> const a= {a: undefined, b: null, c: 1}
undefined
> Object.keys(a)
[ 'a', 'b', 'c' ]
```

Comment (koladilip) [reply src/v0/destinations/braze/util.js:292]: Should we also handle undefined cases?

Comment (koladilip) [reply src/v0/destinations/braze/util.js:292]: that is for storedUserData what about `userData[key] === null`

**Review by koladilip — DISMISSED** [review]

**Pull request merged**